core/jobs/batch_jobs/state_validation_jobs.py

```diff
@@ -0,0 +1,133 @@
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+        )
+
+        exploration_with_invalid_states = (
+            total_explorations
+            | 'Combine exploration id and states' >> beam.Map(
+                lambda exp: (exp.id, exp.states))
+            | 'Get only states' >> beam.Map(
+                lambda objects: (objects[0], self.get_states(objects[1]))
+            )
+            | 'Get invalid states' >> beam.Filter(
+                lambda objects: self.check_invalid(objects[1])
+            )
+            | 'Remove empty string' >> beam.Map(
+                lambda objects: (objects[0], self.remove_empty(objects[1]))
+            )
+        )
+
+        report_number_of_invalid_states_queried = (
+            exploration_with_invalid_states
+            | 'Report count of exp models' >> (
+                job_result_transforms.CountObjectsToJobRunResult('STATES'))
+        )
+
+        report_invalid_states = (
+            exploration_with_invalid_states
+            | 'Save info on invalid exps' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                    'The id of exp is %s and the erroneous states are %s'
+                    % (objects[0], objects[1])
+                ))
+        )
+
+        return (
+            (
+                report_invalid_states,
+                report_number_of_invalid_states_queried
+            )
+            | 'Combine results' >> beam.Flatten()
+        )
+
+    def get_states(self, states_dict):
+        """Returns the array of state content html field
+
+        Args:
+            states_dict: dict. A dictionary of states
+
+        Returns:
+            state_content_html: array. Array containing state
+            content html field
+        """
+        state_content_html = []
+        for key, value in states_dict.items():
+            soup = bs4.BeautifulSoup(value.content.html, 'html.parser')
+            links = soup.find_all('oppia-noninteractive-image')
+            for link in links:
+                state_content_html.append(
+                    (key, link.get('filepath-with-value').replace('&quot;', ''))
+                )
+        return state_content_html
+
+    def check_invalid(self, states):
+        """Checks if the stat is valid or not
+
+        Args:
+            states: list[tuple]. Consist of state name and image value
+
+        Returns:
+            boolean: bool. Returns True if the state is not valid otherwise False
+        """
+        for state in states:
+            if state[1] == '':
+                return True
+        return False
+
+    def remove_empty(self, states):
+        """Removes the empty string which are present with states name
+
+        Args:
+            states: list[tuple]. Consist of state name and empty string
+
+        Returns:
+            states_list: list. All the erroneous states name
+        """
```

Comment:
This only seems to extract the key (state name) and not to remove empty.

Reply:
Done